### DISCUSSION TITLE
404 error when authenticating with password

### DISCUSSION THREAD

**Discussion by devbyVeriTech** [pockethost/pockethost#405] [Q&A] [answered]
I’m having status: 404,
data:{ code: 404, message: ‘Not found.’, data: {}} when trying to authenticatewithpassword.

**Comment by benallfree** [comment]
Are you trying to access your instance's admin? If so, the admin URL is `/_`. Or you can access by using the link from the PocketHost dashboard.

**Reply by devbyVeriTech** [reply]
The issue is not about the admin.

I’m talking about authWithPassword authentication on my website.
If I use localhost url for my pocketbase authentication, it’ll work just fine but since the app is ready for production, I used a pockethost instance and it shows that error in my console log and pockethost log also

**Reply by benallfree** [reply]
Please paste the code that isn't working and the corresponding PocketHost logs.

**Reply by devbyVeriTech** [reply]
export const actions = {
    login: async ({locals, request}) => {
        const body = Object.fromEntries(await request.formData()) as{
          email: string,
          password: string
      }

        try {
          await locals.pb.collection('users').authWithPassword(body.email, body.password);
        } 
        catch (e){
          console.error(e);
          throw e;
      }

      throw redirect (303, '/');
  }
}

**Comment by devbyVeriTech** [comment]
[log_20241024101627869Z.json](https://github.com/user-attachments/files/17511392/log_20241024101627869Z.json)
This is the error from pockethost

**Reply by benallfree** [reply]
I’m sorry that doesn’t look like a PocketHost log. The code you provided also doesn’t specifically show a connection to PocketHost, can you please provide some Chrome network debugger logs so I can confirm exactly what request and response is happening with PocketHost?

You can also try Discord for faster troubleshooting.

**Reply by devbyVeriTech** [reply]
![45164240-B3B1-4CD4-903F-5BBDD2D9DF8C](https://github.com/user-attachments/assets/52ecaef9-abd1-495c-a19a-2662bf8570ea)

this is the screenshot of the log you said doesn’t look like pockethost’s log.

and why you can’t see the connection in the code is because I used hooks.server to import the url from .env then I used +page.server for the authentication logic.
The code I sent is from the+page.server.

the thing here is, if it isn’t connected to pockethost instance, I won’t even see the log in my pockethost instance

**Reply by benallfree** [reply]
That is the pocketbase log, not the pockethost log. The pockethost log is found in the dashboard at pockethost.io

I do not know why your API paths are not working. Please create a repo that is a minimal demonstration of the problem and I’ll take a look :)

**Reply by devbyVeriTech** [reply]
Thanks.
I’ll try to do that but please take a look at the address bar in the screenshot I attached now and the error displayed.
![2CB22130-393E-4FFB-9848-4ED2E4DF7D99](https://github.com/user-attachments/assets/d63a3170-89e5-4bda-8f6c-194a4f04d86f)

**Reply by benallfree** [reply]
That is the admin of a PocketBase instance hosted on PocketHost. PocketHost itself has a dashboard with logs: https://pockethost.io/docs/logs

I’m sorry you’re having trouble, I can imagine I might feel frustrated. The best thing to do is create a minimal repo that shows the problem. Then I can look at the exact code and help. Without the exact code, your problem could be:

* out of date JS client or mismatched PocketBase version
* A JS hook is blocking an API path
* wrong instance url

**Reply by devbyVeriTech** [reply]
[codes.txt](https://github.com/user-attachments/files/17522830/codes.txt)

this txt file contains all the necessary codes.
Please view and give me feedback.
Like I said, I was using the 127.0.0.1(localhost) during development and everything works well.
So I created a pockethost instance so I can deploy the project and I started having the issue even before deploying it.
If I switch the url back to localhost, it’ll continue working perfectly.

**Reply by benallfree** [reply]
This is not exact code. I need a working repo of exact code that reproduces the problem. Code samples won’t be enough.

**Reply by devbyVeriTech** [reply]
https://github.com/devbyVeriTech/sample

repo

**Reply by benallfree** [reply]
Ok, I see. 

Calling PocketBase from server-side is generally an antipattern.

Everything in `page.server.ts` should be called client-side. There is no need for this file at all. Likewise, `+layout.server.ts` is not needed. Instead, use a `writable()` store on the frontend.

If you make those changes, `+hooks.*` goes away too.

Note: you're using `http://` and PocketHost instances use `https://`

**Reply by devbyVeriTech** [reply]
Honestly, I’m not familiar with this new method.
You mean I should write the login logic on my +page.svelte and get rid of the +page.server.ts???

I’ve been using the server-side rendering on all my projects and it works just fine.

can you please send a sample of the writable store in the frontend???

**Reply by devbyVeriTech** [reply]
I can send you another repo where the authentication is done on the server-side.

and I’m using a pockethost instance

**Reply by benallfree** [reply, marked as answer]
> You mean I should write the login logic on my +page.svelte and get rid of the +page.server.ts???

Yes. PocketBase is intended for SPA apps. The PocketBase JS client is generally meant to be used client-side, called from your frontend. Calling PocketBase inside SSR is usually not necessary and indicates an antipattern.

See [Svelte stores](https://svelte.dev/docs/svelte/stores) for managing client-side stores.

You can also review [LoginForm.svelte](https://github.com/pockethost/pockethost/blob/v0/main/packages/dashboard/src/routes/(static)/get-started/LoginForm.svelte#L36) from PocketHost's own frontend (written in SvelteKit) for an example of doing a login client-side.

**Reply by devbyVeriTech** [reply]
Thank you.
I’ll try just that and get back to you.

**Reply by devbyVeriTech** [reply]
But can I use hooks.client.ts ???

**Reply by benallfree** [reply]
Yes, `hooks.client.ts` is fine as it executes client-side only. They are also not needed, though.

**Reply by devbyVeriTech** [reply]
I used the svelte stores as suggested.
I also moved the logic to frontend.
Used hooks.client.ts and everything works perfectly well.

thank you ❤️
